enhancements/operator-framework-api/catalogue-check.md

```diff
@@ -0,0 +1,377 @@
+
+> For the same reasons, SDK would not be ideal add specific OpenShift requirements in SDK. However, if the items can be used by different vendors or services, or indeed be "optional" then shows that could be acceptable.
+
+5. Knowing that:
+- we have not specific implementations in the APIs raised above and provide in SDK or requirements/use cases currently that can let us confirm that it will grow a lot to starts to cause problems or any educated guess to know how long time it could take.
+- if we decide to have a downstream repo for [operator-framework/api][oper-api] and need to keep maintained the feature in [ocp-release-operator-sdk] only.
+
+a) How much extra effort will need to be spend to do the downstream releases and keep these repositories/solutions maintained? Is this effort justifiable by the concerns over use the upstream libs currently?
+
+b) The mainly general concerns addressed via the Validators, *linter* checks, will be available via the new checks provided to OperatorHub when the EP [operator-framework/api: add new checks to the operatorhub validator](github.com/operator-framework/enhancements/pull/65) be implemented. In this way, besides this idea be very interesting for the [audit][audit-ep] idea, CVP/Pipeline, and operator authors which provide solutions for OCP Catalog, if we reach a consensus here to we would need to persuade an alternative solution which brings high effort to get this new Validator implemented and/or maintained then; how much value X effort will this proposal bring? Would still valid we try to address these requirements?
+
+>This shows not exactly about the specific checks but more about the longer term version of the validation framework that allows these kinds of contributions.
+
+## Drawbacks
+
+We might use this scenario as use case to know how we will address this kind of requirement only for downstream in the future.
+
+However, if we check that the effort/cost to address this needs in the possible agreed design will be too high which not justify its value at the moment then, we might decide to not move forward with now.
+
+### Test Plan
+
+The rules implemented should be covered by unit tests.
+
+## Alternatives
+
+### (Option A) Include the OpenShift Validator in upstream
+
+The proposal above was defined considering the new Validator be implemented in [operator-framework/api][oper-api] and used by SDK upstream implementation.
+
+**Pros**
+
+- Lower effort/cost since it would be similar to the others as it usage
+- Allow the Validator be used with OKD
+- The Validator can be used by any solution
+- Does not requires the usage/installation of a downstream SDK binary
+
+**NOTE** SDK users and Pipeline uses the upstream binary and shows that we have not a release process for SDK binary from downstream. Just for the images which are useful for Ansible/Helm operators.
```

Comment:
That is not true. We release a downstream SDK binary now as of 4.7. 

The binary is hosted on the mirror: https://mirror.openshift.com/pub/openshift-v4/clients/operator-sdk/latest/
We are also building an SDK image, albeit solely for giving access to the binary. https://github.com/openshift/ocp-release-operator-sdk/tree/master/release/sdk

Reply:
I just discovered that with you because of another case. I need to update it here. Thank you.

Reply:
solved.